lib/WP_Auth0_Lock10_Options.php

```diff
@@ -113,9 +118,14 @@ protected function build_settings( $settings ) {
 
 		$options_obj['socialButtonStyle'] = 'big';
 
-		if ( isset( $settings['gravatar'] ) && empty( $settings['gravatar'] ) ) {
+		if ( isset( $settings['gravatar'] ) && '' !== $settings['gravatar'] && empty( $settings['gravatar'] ) ) {
```

Comment:
empty is already "". If you check for `!=""` and `empty` at the same time, this will never be called! What is going on here? 🤕

```
'' !== $settings['gravatar'] && empty( $settings['gravatar'] )
```

Reply:
It's a little wonky because it has a number of settings arrays that override each other with successive option building. 

`''` is an empty string (in this case, means use the default), `isset()` is whether that key exists or not (avoids `Undefined index` errors), and `empty()` checks for `0`, `null`, `false`, etc.